frontend/src/lib/components/ReverseProxyChecker/reverseProxyCheckerLogic.ts

```diff
@@ -13,17 +13,25 @@ const CHECK_INTERVAL_MS = 1000 * 60 * 60 // 1 hour
 
 export const reverseProxyCheckerLogic = kea<reverseProxyCheckerLogicType>([
     path(['components', 'ReverseProxyChecker', 'reverseProxyCheckerLogic']),
-    loaders({
+    loaders(({ values, cache }) => ({
         hasReverseProxy: [
             false as boolean | null,
             {
                 loadHasReverseProxy: async () => {
+                    if (cache.lastCheckedTimestamp > Date.now() - CHECK_INTERVAL_MS) {
+                        return values.hasReverseProxy
+                    }
+
+                    cache.lastCheckedTimestamp = Date.now()
+
                     const query: HogQLQuery = {
                         kind: NodeKind.HogQLQuery,
-                        query: hogql`SELECT properties.$lib_custom_api_host AS lib_custom_api_host
+                        query: hogql`SELECT DISTINCT properties.$lib_custom_api_host AS lib_custom_api_host
                                 FROM events
                                 WHERE timestamp >= now() - INTERVAL 1 DAY 
                                 AND timestamp <= now()
+                                AND properties.$lib_custom_api_host IS NOT NULL
+                                AND event = '$pageview'
```

Comment:
nit picking that this won't work if you have a mobile app 🙈 

you could do `event in [$pageview, $screen]` or similar